storage/adapters/pgp_adapter.js

```diff
@@ -0,0 +1,198 @@
+/*
+ * Copyright © 2018 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ */
+
+'use strict';
+
+const path = require('path');
+const Promise = require('bluebird');
+const monitor = require('pg-monitor');
+const pgpLib = require('pg-promise');
+const QueryFile = require('pg-promise').QueryFile;
+const BaseAdapter = require('./base_adapter');
+
+const resultCountToMethodMap = {
+	0: 'none',
+	1: 'one',
+	3: 'any',
+	[undefined]: 'any',
+};
+
+class PgpAdapter extends BaseAdapter {
+	/**
+	 *
+	 * @param {Object} options
+	 * @param {Boolean} options.inTest
+	 * @param {string} options.logger
+	 * @param {string} options.sqlDirectory
+	 */
+	constructor(options) {
+		super({
+			engineName: 'pgp',
+			inTest: options.inTest,
```

Comment:
what is the purpose of this `inTest`?

Reply:
It's only used for pg-promisse`noLocking` flag for now.
Not sure if it can be useful for other adapters though.

Reply:
Maybe it's better to pass the separate Postgres option object, which would overwrite the default Postgres flags.

Reply:
I also have some suggestions to make regarding the adapter that I would like to discuss but I reckon we should wait til @nazarhussain is back to discuss those points.

My suggestion is to merge without changing this and create an issue after the discussion if need be.